server/blob/blob-union/src/main/java/org/apache/james/blob/union/UnionBlobStore.java

```diff
@@ -105,6 +105,16 @@ private UnionBlobStore(BlobStore currentBlobStore, BlobStore legacyBlobStore) {
         }
     }
 
+    @Override
+    public BucketName getDefaultBucketName() {
```

Comment:
I think this is an issue...

IMO clearly if the values are different we run into unpredictable behaviour...

```
B1 is the default in currentBlobStore
B2 is the default in legacyBlobStore

The user wants to read mail maim m1 - stored in legacy blob store default bucket B2.

The caller reads the default bucket which will always be B1! and will not find m1 in the legacy blob store as he will attempt a read with B1 while the mail is stored in B2
```

A temporary solution might be to ensure both default buckets are the same and return the common value...